docarray/array/array_stacked.py

```diff
@@ -90,11 +90,23 @@ def from_iterable_document(
                 docs, tensor_type=self.tensor_type
             )
         )
+        self._nested_stack(self._docs)
         self.tensor_type = self._docs.tensor_type
         self._doc_columns, self._tensor_columns = self._create_columns(
             self._docs, tensor_type=self.tensor_type
         )
 
+    def _nested_stack(self: T, docs: DocumentArray):
```

Comment:
I believe you created this function because the DA field will not be collect in `_get_columns_schema` right ? If yes I believe we might just drop this `_get_columns_schema` and merge your for loop with the one inside `_create_columns`.

smth like this

```python
for field, type_ in cls.__fields__.items():
    if issubclas(type_, AbstractTensor):
        ...
    elif issublcas(type_, BaseDocument):
        ...
    elif issubclass(type_, DocumentArray):
        ... # here this is the content of this PR
    else:
        pass
```